pkg/scheduler/api/devices/nvidia/vgpu4pd/utils.go

```diff
@@ -0,0 +1,438 @@
+		Name:   name,
+		Device: make(map[int]*GPUDevice),
+	}
+	for index, val := range tmp {
+		if strings.Contains(val, ",") {
+			items := strings.Split(val, ",")
+			count, _ := strconv.Atoi(items[1])
+			devmem, _ := strconv.Atoi(items[2])
+			health, _ := strconv.ParseBool(items[4])
+			i := GPUDevice{
+				ID:     index,
+				UUID:   items[0],
+				Number: uint(count),
+				Memory: uint(devmem),
+				Type:   items[3],
+				Health: health,
+			}
+			retval.Device[index] = &i
+		}
+	}
+	return retval
+}
+
+func encodeContainerDevices(cd []ContainerDevice) string {
+	tmp := ""
+	for _, val := range cd {
+		tmp += val.UUID + "," + val.Type + "," + strconv.Itoa(int(val.Usedmem)) + "," + strconv.Itoa(int(val.Usedcores)) + ":"
+	}
+	klog.V(4).Infoln("Encoded container Devices=", tmp)
+	return tmp
+	//return strings.Join(cd, ",")
+}
+
+func encodePodDevices(pd []ContainerDevices) string {
+	var ss []string
+	for _, cd := range pd {
+		ss = append(ss, encodeContainerDevices(cd))
+	}
+	return strings.Join(ss, ";")
+}
+
+func decodeContainerDevices(str string) ContainerDevices {
+	if len(str) == 0 {
+		return ContainerDevices{}
+	}
+	cd := strings.Split(str, ":")
+	contdev := ContainerDevices{}
+	tmpdev := ContainerDevice{}
+	//fmt.Println("before container device", str)
+	if len(str) == 0 {
+		return contdev
+	}
+	for _, val := range cd {
+		if strings.Contains(val, ",") {
+			//fmt.Println("cd is ", val)
+			tmpstr := strings.Split(val, ",")
+			tmpdev.UUID = tmpstr[0]
+			tmpdev.Type = tmpstr[1]
+			devmem, _ := strconv.ParseInt(tmpstr[2], 10, 32)
+			tmpdev.Usedmem = int32(devmem)
+			devcores, _ := strconv.ParseInt(tmpstr[3], 10, 32)
+			tmpdev.Usedcores = int32(devcores)
+			contdev = append(contdev, tmpdev)
+		}
+	}
+	//fmt.Println("Decoded container device", contdev)
+	return contdev
+}
+
+func decodePodDevices(str string) []ContainerDevices {
+	if len(str) == 0 {
+		return []ContainerDevices{}
+	}
+	var pd []ContainerDevices
+	for _, s := range strings.Split(str, ";") {
+		cd := decodeContainerDevices(s)
+		pd = append(pd, cd)
+	}
+	return pd
+}
+
+func checkVGPUResourcesInPod(pod *v1.Pod) bool {
+	for _, container := range pod.Spec.Containers {
+		_, ok := container.Resources.Limits[VolcanoVGPUMemory]
+		if ok {
+			return true
+		}
+		_, ok = container.Resources.Limits[VolcanoVGPUNumber]
+		if ok {
+			return true
+		}
+	}
+	return false
+}
+
+func resourcereqs(pod *v1.Pod) []ContainerDeviceRequest {
+	resourceName := v1.ResourceName(VolcanoVGPUNumber)
+	resourceMem := v1.ResourceName(VolcanoVGPUMemory)
+	resourceMemPercentage := v1.ResourceName(VolcanoVGPUMemoryPercentage)
+	resourceCores := v1.ResourceName(VolcanoVGPUCores)
+	counts := []ContainerDeviceRequest{}
+	//Count Nvidia GPU
+	for i := 0; i < len(pod.Spec.Containers); i++ {
+		singledevice := false
+		v, ok := pod.Spec.Containers[i].Resources.Limits[resourceName]
+		if !ok {
+			v, ok = pod.Spec.Containers[i].Resources.Limits[resourceMem]
+			singledevice = true
+		}
+		if ok {
```

Comment:
It would look neater with `else`

Reply:
the logic won't be the same if I use (else) instead of (if ok)